### PR TITLE
[SYCL] updating tests for SYCL2020 errc conformance

### PR DESCRIPTION
**Pull request by cperkinsintel**
in https://github.com/intel/llvm/pull/5582 we are adding an update that ensures that the errc of any exception is correct as specified in the SYCL2020 spec. Here we are updating and expanding the tests to accompany

Signed-off-by: Chris Perkins <chris.perkins@intel.com>

### CODE REVIEW

File: SYCL/Basic/buffer/buffer.cpp [hunk 1]
```diff
@@ -394,7 +394,7 @@ int main() {
       myQueue.submit([&](handler &cgh) {
         accessor<int, 2, access::mode::write, access::target::device,
                  access::placeholder::false_t>
-            B(Buffer, cgh, range<2>(20, 20), id<2>(10, 10));
+            B(Buffer, cgh, range<2>(10, 10), id<2>(10, 10));
```
Comment (againull) [inline SYCL/Basic/buffer/buffer.cpp:397]: These changes look unrelated to SYCL2020 errc conformance. Can we remove them from this PR?

File: SYCL/Basic/buffer/subbuffer.cpp [hunk 2]
```diff
@@ -3,6 +3,9 @@
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
 // RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// UNSUPPORTED: (opencl && gpu)
```
Comment (againull) [inline SYCL/Basic/buffer/subbuffer.cpp:7]: All changes in this file doesn't look related to SYCL2020 errc conformance, I couldn't find any errc checks in the file.
It is possible to move them to a separate PR if they are unrelated? Does it make sense to create a separate test for your modified test case (which shows a problem on opencl gpu) to avoid disabling whole test for opencl gpu?

File: SYCL/Basic/buffer/buffer.cpp (hunk 1 above)
Comment (cperkinsintel) [reply SYCL/Basic/buffer/buffer.cpp:397]: One of the conformance checks is that out-of-bounds range args should trigger an exception in accessor construction. I added that.  Without this change, this test triggers that error.

File: SYCL/Basic/buffer/subbuffer.cpp (hunk 2 above)
Comment (cperkinsintel) [reply SYCL/Basic/buffer/subbuffer.cpp:7]: This test was using an out-of-bounds range argument to an accessor. And, oddly, that was masking an error that is occurring over on OCL:GPU. I reduced that down and reported it, there is a ticket against it now. But in the interim, this test won't pass on OCL:GPU, not due to any bug in the error system, but due to a real bug that OCL:GPU has handling offsets.  Strangely, that is the only device with the problem. L0, OCL:CPU, OCL:ACC and Host are all fine.

**Review by againull — APPROVED** [review]
